### PR TITLE
own: in ownership search results, add link to result to modify query to `file:has.owner($person)`

### PR DESCRIPTION
**Pull request by limitedmage**
Closes https://github.com/sourcegraph/sourcegraph/issues/48837

In ownership search results, adds a link to each search result to modify query to remove `select:file.owners` and add `file:has.owner($person)`. The email or handle returned from the PersonMatch or TeamMatch will be used (not the Sourcegraph username/team name).

This also removes the hover style from the gray box in all search results where the box is not clickable (repo, file path, owner; still present in diff/commit).

## Test plan

https://user-images.githubusercontent.com/206864/223870371-335043ed-3b25-42e9-8007-a83efca23213.mp4

## App preview:

- [Web](https://sg-web-jp-linkfilessearch.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (+0.11 kb) | 0.01% (+1.47 kb) | 0.01% (+1.36 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/jp/linkfilessearch/compare-report.html;tab=live_object) for a full comparison between the commits f493da3 and 599c4fc or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/branded/src/search-ui/components/OwnerSearchResult.tsx [hunk 1]
```diff
@@ -106,9 +140,14 @@ export const OwnerSearchResult: React.FunctionComponent<PersonSearchResultProps>
...
                 data-testid="owner-search-result"
             >
-                <div className={resultStyles.matchType}>
-                    <small>Owner match</small>
-                </div>
+                <small className={resultStyles.matchType}>
+                    <span>Owner match</span>
+                    {fileSearchLink && (
+                        <Link to={fileSearchLink} className={styles.filesLink}>
+                            Which files?
```
Comment (eseliger) [inline client/branded/src/search-ui/components/OwnerSearchResult.tsx:147]:
```suggestion
                        <Link to={fileSearchLink} className={styles.filesLink}>
                            Show files
```

maybe?

**Review by eseliger — APPROVED** [review]
such a good addition!

**Review by leonore — APPROVED** [review]
🔥🔥🔥

File: client/branded/src/search-ui/components/OwnerSearchResult.tsx (hunk 1 above)
Comment (limitedmage) [reply client/branded/src/search-ui/components/OwnerSearchResult.tsx:147]: Yeah I like this wording better

**Pull request merged**